### ISSUE TITLE
Pull Request Triggers

### ISSUE THREAD

**Issue by SourceStudio** [Azure/azure-cli#9982]
I created an ACR task which successfully fires on a push to the master branch of a GitHub repository, but I could not seem to get it to fire on the opening of a pull request.  Are pull request triggers able to work with private GitHub repositories? Are there any other conditions that need to be satisfied before the trigger fires, or should it simply require a pull request to be opened? Thanks.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: bd0e3f3c-5aed-28b6-1e28-22ff32a8521d
* Version Independent ID: 128e5bc6-e518-bd5a-f511-48e0ed185739
* Content: [az acr task](https://docs.microsoft.com/en-us/cli/azure/acr/task?view=azure-cli-latest#feedback)
* Content Source: [src/azure-cli/azure/cli/command_modules/acr/_help.py](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acr/_help.py)
* Service: **container-registry**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by northtyphoon** [comment]
@SourceStudio we have disabled Pull Request trigger due to some security concerns and plan to remove it.

**Comment by northtyphoon** [comment]
@jaysterp

**Comment by jaysterp** [comment]
@SourceStudio The trigger would be fired whenever a pull request is `assigned`, `unassigned`, `labeled`, `unlabeled`, `opened`, `edited`, `closed`, `reopened`, `synchronize`, `ready_for_review`, `locked`, `unlocked` or when a pull request review is requested or removed. As mentioned above, the feature is disabled currently. We are very interested in learning more about how customers might use this type of git trigger. Would you mind sharing your use case scenario? 😃

**Comment by SourceStudio** [comment]
Thanks @northtyphoon @jaysterp for your responses. We intended to use the pull request trigger to automate the building of feature branches in some of our internal repositories.  The resulting container images would have been automatically deployed to new Web App instances as part of the ACR build task. We planned to use the timer trigger to automate the removal of old Web Apps as necessary. Hope this helps.

**Comment by jaysterp** [comment]
@SourceStudio Thank you for the explanation. Will the existing ACR Task git commit trigger feature apply to your scenario? That trigger will fire whenever a commit is made to your source branch (including when a PR is merged). Sounds like a great use for timer triggers 😃

**Comment by SourceStudio** [comment]
@jaysterp, thanks for your response. We could make it work, but since our feature branches sometimes live for a few weeks before being merged into master, it would have been more useful to have been able to open a pull request and have ACR continuously build such branches until they were ready to be moved to production. We could probably alter our workflow slightly so we create an ACR build task for each feature branch manually as necessary, but this is obviously not as convenient as pull request triggers would have been.

**Comment by jaysterp** [comment]
Hi @SourceStudio. Thank you for your feedback. Our initial concern with the pull request trigger was because it was default to enabled in the CLI. This provided a security concern because any pull request opened, which may or may not contain malicious code, would trigger a image build. Therefore, we decided to block the feature server side until we saw a solid scenario to support. However, after discussion with the team, we see there is no concern if we have this trigger type defaulted to disabled with users opting in via the CLI if they so wish. We see for scenarios such as the one you described, the pull request trigger is more optimal than the commit trigger. This feature will once again be supported pending a rollout. We'll let you know when it's available. Thanks!

**Comment by SourceStudio** [comment]
Thanks for your response @jaysterp.  That's great news, and will greatly simplify things in our case!  One other potential solution to this would perhaps be to allow multiple branches to be specified in the commit trigger of the ACR build task, or even allow a regular expression to be defined that indicates which branches are authorized to be built as part of the task.  Of course, re-enabling the pull-request trigger is the best solution for our scenario, but allowing multiple branches may provide an additional level of flexibility.

**Comment by jaysterp** [comment]
@SourceStudio thank you for the feedback for incorporating multi-branch support. We will discuss this with the team. The roll out is complete and pull request triggers are now enabled. Can you please create a new task with `--pull-request-trigger-enabled true` and let us know if you are unblocked? Thanks!

**Comment by SourceStudio** [comment]
Hi @jaysterp, apologies for the late reply.  I've just tested this functionality and it works as we had expected - many thanks for your help in resolving this.
